back/pkg/roomcontroller/models/roomserviceprovider.go

```diff
@@ -9,5 +9,5 @@ type RoomServiceProvider interface {
 	GetRoom(name string) (entityservice.Room, error)
 	AddRoom(room entityservice.Room) error
```

Comment:
Ich glaube wir sollten den Room Controller mal umbenennen oder auseinander ziehen.
Ich dachte dann evtl. noch an weitere Subrouter, wenn wir z.B. eine API haben mit `/<building>/<room>/` oder `/<building>/<locations>/` oder `/<building>/<map>/...` usw. 

Also übergeordnet einen Building Router und dann die verschiedenen Subrouter darunter für locations, map und rooms.

Reply:
Ich denke auch dass wir, wenn wir mehr in Richtung Users gehen, das auch brauchen werden. Da bekommen wir bestimmt noch ein paar andere APIs dazu.

Reply:
> Also übergeordnet einen Building Router und dann die verschiedenen Subrouter darunter für locations, map und rooms.

Ja, irgendwie so in der art...

Ich bin mir noch nicht ganz sicher wie das aussehen soll.
Verschiedene resources sollten schon verschiedene URL einstiegspunkte haben.

.......roomapi
.......roomapi/roooms
.......roomapi/roooms/:name
.......roomapi/roooms?filterkram=filer&....

..locationapi
..locationapi/locations
..locationapi/locations/:name
..locationapi/locations?filterkram=filer&....

..........mapapi
..........mapapi/mapitems
..........mapapi/mapitems/:name
..........mapapi/mapitems?filterkram=filer&....
------------------------------------------------------------

Bei den calls bin ich mir nicht sicher wie wir die unterbringen sollen....


......roomapi/building/:building/floor/:floor/rooms
..locationapi/building/:building/floor/:floor/locations
..........mapapi/building/:building/floor/:floor/mapitems